extensions/resource-deployment/src/services/azdataService.ts

```diff
@@ -138,8 +133,20 @@ export class AzdataService implements IAzdataService {
 		});
 	}
 
-	private getJsonObjectFromFile(path: string): any {
-		return JSON.parse(fs.readFileSync(path, 'utf8'));
+	private getJsonObjectFromFile(path: string): Thenable<any> {
```

Comment:
I recommend using Promise instead of Thenable for these. Thenable was added as a compatibility layer and if we don't need the compatibility with other Thenables here we should use promise

Reply:
agreed, for private methods I am using Promise now.